oas.go

```diff
@@ -136,7 +136,7 @@ type SchemaObject struct {
 	FieldName          string              `json:"-"` // For goas
 	DisabledFieldNames map[string]struct{} `json:"-"` // For goas
 
-	Type                 string                 `json:"type,omitempty"`
+	Type                 *string                `json:"type,omitempty"`
```

Comment:
If `type` is `""` it should get omitted from the json response. Was this not the case?

Reply:
Right but we don't need it omitted, we need a value of `type: {}` so that OpenAPI interprets that as any value.

Reply:
ah gotcha! so setting it `nil` will output `type: {}`? I didn't see this in the test example, would you mind adding it?

Reply:
My bad, it isn't `type: {}` it's the `key: {}` you can see it here: https://github.com/launchdarkly/goas/pull/15/files#diff-eff279f70f1a81e2abdae4e57736a72120ce3a184bc25ccdbbb60cb10b3027c5L193-R192

Reply:
oh okay cool, does this mean `Type` no longer needs to be a pointer then?

Reply:
no, that's what I ended up needed for it to accept any value. For example now any flag type works for the generated clients I've tested so far: Go and Python. Without this pointer change they will not accept anything except a json flag.